cpp/net/url_fetcher.cc

```diff
@@ -174,18 +189,19 @@ void State::RequestDone(evhttp_request* req) {
   }
```

Comment:
Leaking `req` when returning inside this block? Maybe use a `unique_ptr` as a deleter?